Makefile

```diff
@@ -2,11 +2,11 @@ init:
 	pip install -r requirements.txt
 devel:
 	pip install -r requirements_dev.txt
-	pre-commit install
+	pre-commit install --hook-type pre-commit --hook-type pre-push
 test:
 	tox
-lint:
-	flake8 pyetrade tests
+analysis: # Lint, format, import optimizer, etc.
+	pipenv run pre-commit run --all-files

```

Comment:
I would like not to use `pipenv` given the choice. I am not a big fan of the project and the same effect could be accomplished by getting into the venv and running `pre-commit run --all-files`. In the future I might change to something like `poetry` because it uses a little more standard  PEP conventions of just using the `pyproject.toml`.

Reply:
oh i see, no problem. I'm not too familiar with `poetry` but i would be willing to give it a try and set it up for the project (if you don't mind that is).

Also to be clear, this is what you are referring to correct?: https://python-poetry.org/

Reply:
I have removed `pipenv` and replaced with `poetry`. Please see my latest comment and commits. Thank you.